applications/dds_video/dds_video.cpp

```diff
@@ -0,0 +1,259 @@
+};
+
+/**
+ * @brief Application to render a DDS video stream (published by the DDSVideoPublisher)
+ * and shapes (published by the RTI Connext Shapes Demo) to Holoviz.
+ */
+class DDSToHoloviz : public holoscan::Application {
+ public:
+  explicit DDSToHoloviz(uint32_t domain_id, uint32_t stream_id)
+      : domain_id_(domain_id), stream_id_(stream_id) {}
+
+  void compose() override {
+    using namespace holoscan;
+
+    std::shared_ptr<UnboundedAllocator> allocator = make_resource<UnboundedAllocator>("pool");
+
+    //  DDS Video Subscriber
+    auto participant_qos = std::string("HoloscanDDSTransport::SHMEM+LAN");
+    auto video_subscriber = make_operator<ops::DDSVideoSubscriberOp>("video_subscriber",
+        Arg("allocator", allocator),
+        Arg("domain_id", domain_id_),
+        Arg("stream_id", stream_id_),
+        Arg("participant_qos", participant_qos),
+        Arg("reader_qos", std::string("HoloscanDDSDataFlow::Video")));
+
+    // DDS Shapes Subscriber
+    auto shapes_subscriber = make_operator<ops::DDSShapesSubscriberOp>("shapes_subscriber",
+        Arg("domain_id", domain_id_),
+        Arg("participant_qos", participant_qos),
+        Arg("reader_qos", std::string("HoloscanDDSDataFlow::Shapes")));
+
+    // DDS Shapes Renderer
+    auto shapes_renderer = make_operator<ops::DDSShapesRenderer>("shapes_renderer",
+        Arg("allocator", allocator));
+
+    // Holoviz (initialize with the default input spec for the video stream)
+    std::vector<ops::HolovizOp::InputSpec> input_spec;
+    auto& video_spec = input_spec.emplace_back(
+        ops::HolovizOp::InputSpec("", ops::HolovizOp::InputType::COLOR));
+    auto holoviz = make_operator<ops::HolovizOp>("holoviz",
+        Arg("width", 640u), Arg("height", 480u), Arg("tensors", input_spec));
+
+    add_flow(video_subscriber, holoviz, {{"output", "receivers"}});
+    add_flow(shapes_subscriber, shapes_renderer, {{"output", "input"}});
+    add_flow(shapes_renderer, holoviz, {{"outputs", "receivers"}});
+    add_flow(shapes_renderer, holoviz, {{"output_specs", "input_specs"}});
+  }
+
+ private:
+  uint32_t domain_id_;
+  uint32_t stream_id_;
+};
+
+int main(int argc, char** argv) {
+  bool publisher = false;
+  bool subscriber = false;
+  uint32_t stream_id = 0;
+  uint32_t domain_id = 0;
+
+  struct option long_options[] = {
+      {"publisher", no_argument, 0, 'p'},
+      {"subscriber", no_argument, 0, 's'},
+      {"id", required_argument, 0, 'i'},
+      {"domain", required_argument, 0, 'd'},
+      {0, 0, 0, 0}};
+
+  while (true) {
+    int option_index = 0;
+
+    const int c = getopt_long(argc, argv, "psi:d:", long_options, &option_index);
+    if (c == -1) { break; }
+
+    const std::string argument(optarg ? optarg : "");
+    switch (c) {
+      case 'p':
+        publisher = true;
+        break;
+      case 's':
+        subscriber = true;
+        break;
+      case 'c':
+        stream_id = stoi(argument);
+        break;
+      case 'd':
```

Comment:
Could you please document these CLI options in the README or via a default app printout?

Reply:
Added a `--help`/`usage()` to the application.